lib/convection/model/template.rb

```diff
@@ -183,8 +198,121 @@ def diff(other, stack_ = nil)
       end
 
       def to_json(stack_ = nil, pretty = false)
-        return JSON.generate(render(stack_)) unless pretty
-        JSON.pretty_generate(render(stack_))
+        rendered_stack = render(stack_)
+        validate(rendered_stack)
+
+        return JSON.generate(rendered_stack) unless pretty
+        JSON.pretty_generate(rendered_stack)
+      end
+
+      def validate(rendered_stack = nil)
+        %w(resources mappings parameters outputs description bytesize).map do
+          |method| self.send("validate_#{method}", rendered_stack)
```

Comment:
* Unnecessary use of `self`
* Put the argument list (`|...|`) on the same line as `do`/`{`